src/targets/maven.ts

```diff
@@ -0,0 +1,389 @@
+   */
+  private async extractArtifact(
+    artifact: RemoteArtifact,
+    dir: string
+  ): Promise<void> {
+    this.logger.debug(`Downloading ${artifact.filename}...`);
+    const downloadedPkgPath = await this.artifactProvider.downloadArtifact(
+      artifact
+    );
+    this.logger.debug(
+      `Extracting ${artifact.filename} to ${downloadedPkgPath}...`
+    );
+    await extractZipArchive(downloadedPkgPath, dir);
+  }
+
+  /**
+   * Uploads the given distribution, including all files that are required.
+   *
+   * @param distDir directory of the distribution.
+   */
+  private async uploadDistribution(distDir: string): Promise<void> {
+    const {
+      targetFile,
+      javadocFile,
+      sourcesFile,
+      pomFile,
+    } = this.getFilesForMavenCli(distDir);
+
+    this.retrySpawnProcess(
+      () =>
+        spawnProcess(this.mavenConfig.mavenCliPath, [
+          'gpg:sign-and-deploy-file',
+          `-Dfile=${targetFile}`,
+          `-Dfiles=${javadocFile},${sourcesFile}`,
+          `-Dclassifiers=javadoc,sources`,
+          `-Dtypes=jar,jar`,
+          `-DpomFile=${pomFile}`,
+          `-DrepositoryId=${this.mavenConfig.mavenRepoId}`,
+          `-Durl=${this.mavenConfig.mavenRepoUrl}`,
+          `--settings ${this.mavenConfig.mavenSettingsPath}`,
+        ]),
+      'Uploading'
+    );
+  }
+
+  /**
+   * Retrieves a record of all the required files by Maven CLI to upload
+   * anything.
+   *
+   * @param distDir directory of the distribution.
+   * @returns record of required files.
+   */
+  private getFilesForMavenCli(distDir: string): Record<string, string> {
+    const moduleName = parse(distDir).base;
+    const targetFile = join(distDir, this.getTargetFilename(distDir));
+    const javadocFile = join(distDir, `${moduleName}-javadoc.jar`);
+    const sourcesFile = join(distDir, `${moduleName}-sources.jar`);
+    const pomFile = join(distDir, 'pom-default.xml');
+    return {
+      targetFile,
+      javadocFile,
+      sourcesFile,
+      pomFile,
+    };
+  }
+
+  /**
+   * Retrieves the target file name for the current distribution.
+   *
+   * If the distibution is an Android distribution, the target file is the
+   * file containing "release" in the name and the ".aar" extension.
+   * Typically, the module (directory) name without the version and appending
+   * "-release.aar" at the end.
+   *
+   * If the distribution isn't an Android distribution, the target filename is
+   * the module name appending ".jar" to the end.
+   *
+   * @param distDir directory where distributions are.
+   * @returns the target file name.
+   */
+  private getTargetFilename(distDir: string): string {
+    const moduleName = parse(distDir).base;
+    const isAndroidDistDir = this.config.androidDistDirPattern.test(moduleName);
+    if (isAndroidDistDir) {
+      return moduleName.replace(
+        this.config.androidFileSearchPattern,
+        this.config.androidFileReplaceStr
+      );
```

Comment:
Maybe we can group these Android-related settings under a namespace:
```suggestion
    const isAndroidDistDir = this.config.android.distDirPattern.test(moduleName);
    if (isAndroidDistDir) {
      return moduleName.replace(
        this.config.android.fileSearchPattern,
        this.config.android.fileReplaceStr
      );
```